test_cleanup.yml

```diff
@@ -0,0 +1,105 @@
+---
+# Test script for deployment cleanup functionality
+#
+# HOW TO RUN:
+#   ansible-playbook test_cleanup.yml                   # Basic test (keep 3)
+#   ansible-playbook test_cleanup.yml --check           # Dry run (no changes made)
+#   ansible-playbook test_cleanup.yml -e keep_count=5   # Test keeping 5 deployments
+#
+
+- name: Test deployment cleanup
+  hosts: localhost
+  connection: local
+  gather_facts: true
+  vars:
+    test_dir: "/tmp/ansible_cleanup_test_{{ ansible_date_time.epoch }}"
+    deploy_user: "{{ ansible_user_id }}"
+    deploy_keep_count: "{{ keep_count | default(3) | int }}"
+    
+    # These are our fake deployment versions (oldest to newest)
+    fake_deployments:
+      - "2.1.0-old1"
+      - "2.2.0-old2" 
+      - "2.3.0-old3"
+      - "3.0.0-keep1"
+      - "3.1.0-keep2"
+      - "3.2.0-keep3"
+      - "4.0.0-previous"
+      - "4.1.0-current"
+
+  tasks:
+
+    - name: "Create test directory"
+      file:
```

Comment:
these are not blockers. Can we use ansible fqrn so instead of `find` `ansible.builtin.find`